src/types/mod.rs

```diff
@@ -149,7 +149,10 @@ macro_rules! pyobject_native_type_named (
 #[macro_export]
 macro_rules! pyobject_native_static_type_object(
     ($typeobject:expr) => {
-        |_py| unsafe { ::std::ptr::addr_of_mut!($typeobject) }
+        |_py| {
+            #[allow(unused_unsafe)] // https://github.com/rust-lang/rust/pull/125834
```

Comment:
Ah great, so it looks like one day we won't need the `allow` at all. Great investigation!

Reply:
It's probably gonna be a while until we're at MSRV 1.82, but eventually is better than never 😆